edx_notifications/stores/sql/models.py

```diff
@@ -0,0 +1,256 @@
+        """
+        Generate a NotificationType data object
+        """
+
+        return NotificationUserMap(
+            id=self.id,
+            user_id=self.user_id,
+            msg=self.msg.to_data_object(),  # pylint: disable=no-member
+            read_at=self.read_at,
+            user_context=DictField.from_json(self.user_context)
+        )
+
+    @classmethod
+    def from_data_object(cls, user_map):
+        """
+        create a ORM model object from a NotificationType
+        """
+
+        obj = SQLNotificationUserMap()
+        obj.load_from_data_object(user_map)
+        return obj
+
+    def load_from_data_object(self, user_map):
+        """
+        Hydrate ourselves from a passed in user_map
+        """
+
+        self.id = user_map.id  # pylint: disable=attribute-defined-outside-init
+        self.user_id = user_map.user_id
+        self.msg = SQLNotificationMessage.from_data_object(user_map.msg)
+        self.read_at = user_map.read_at
+        self.user_context = DictField.to_json(user_map.user_context)
+
+
+class SQLNotificationChannel(models.Model):
+    """
+    Information about how notifications are delivered, e.g. web, triggered email,
+    SMS, iOS Push Notifications, etc.
+    """
+
+    class Meta(object):
+        """
+        ORM metadata about this class
+        """
+        app_label = 'edx_notifications'  # since we have this models.py file not in the root app directory
+        db_table = 'edx_notifications_notificationchannel'
+
+
+class SQLNotificationTypeRenderingProvided(models.Model):
+    """
+    Describes which rendering types this notification type supports, e.g. 'json', 'text', 'short-html', 'long-html'
+    """
+
+    class Meta(object):
+        """
+        ORM metadata about this class
+        """
+        app_label = 'edx_notifications'  # since we have this models.py file not in the root app directory
+        db_table = 'edx_notifications_notificationtyperenderingprovided'
+
+
+class SQLNotificationUserTypeChannelMap(models.Model):
+    """
+    User specific mappings of Notifications to Channel, to reflect user preferences
+    """
+
+    class Meta(object):
+        """
+        ORM metadata about this class
+        """
+        app_label = 'edx_notifications'  # since we have this models.py file not in the root app directory
+        db_table = 'edx_notifications_notificationusertypechannelmap'
+
+
+class SQLDisplayString(models.Model):
+    """
+    NOTE: These can be cached completely in memory
+    """
+
+    string_name = models.CharField(max_length=255, db_index=True)
+    lang = models.CharField(max_length=16, db_index=True)
+    string_value = models.TextField()
+
+    class Meta(object):
+
+        """
+        ORM metadata about this class
+        """
+        app_label = 'edx_notifications'  # since we have this models.py file not in the root app directory
+        db_table = 'edx_notifications_displaystring'
+        unique_together = (('string_name', 'lang'),)
```

Comment:
scratch pad stuff, but - in essence - I was thinking through what it might mean to have display strings in a database which might need to support multiple languages. But I'm not sure this will actually be necessary.